swap_meet/vendor.py

```diff
@@ -0,0 +1,109 @@
+
+class Vendor:
+
+    def __init__(self, inventory=None):
+        if inventory == None:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+        
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+        
+
+    def remove(self, item):
+        if item in self.inventory:
+            self.inventory.remove(item)
+            return item
+        else:
+            return False
+
+    def get_by_category(self, category):
+        category_list = []
+        for item in self.inventory:
+            if category == item.category:
+                category_list.append(item)
+        return category_list
+
+    def swap_items(self, other_vendor, my_item, their_item):
+        
+        if their_item not in other_vendor.inventory or my_item not in self.inventory:
+
+            return False     
+
+        else:
+
+            other_vendor.inventory.append(my_item)
+            other_vendor.remove(their_item)
+            self.inventory.remove(my_item)
+            self.inventory.append(their_item)
+
+            return True
+    
+    def swap_first_item(self, other_vendor):
+
+
+        if self.inventory == [] or other_vendor.inventory == []:
+
+            return False
+
+        else:
+
+            other_vendor.inventory.append(self.inventory[0])
+            self.inventory.remove(self.inventory[0])
+            self.inventory.append(other_vendor.inventory[0])
+            other_vendor.inventory.remove(other_vendor.inventory[0])
+
+            return True
```

Comment:
👍  this works perfectly! but let's look at lines 32 - 43 above. We are almost doing the same thing twice, so how could we fix that? We could use the `swap_items` method inside of this one.

```suggestion
def swap_first_item(self, other_vendor):
        return self.swap_items(other_vendor, self.inventory[0], other_vendor.inventory[0])
```

So, we call `swap_items` and pass in the arguments it needs: the other vendor, our item, other vendor's item. So we grab the first item out of the inventories by using index 0. Now we let the method return `True` or `False`